lib/Onyx.js

```diff
@@ -754,7 +754,19 @@ function connect(mapping) {
     // Commit connection only after init passes
     deferredInitTask.promise
         .then(() => addKeyToRecentlyAccessedIfNeeded(mapping))
-        .then(getAllKeys)
+        .then(() => {
+            // Performance improvement
+            // If the mapping is connected to an onyx key that is not a collection
+            // we can skip the call to getAllKeys() and return an array with a single item
+            if (Boolean(mapping.key)
+                && typeof mapping.key === 'string'
+                && !(mapping.key.endsWith('_'))
+                && cache.storageKeys.has(mapping.key)
+            ) {
+                return [mapping.key];
```

Comment:
NAB, but now that we do this - some of the comments below are a little misleading. Since this is not a key that we found (we just assumed it exists).